modules/veteran/app/controllers/veteran/v0/accredited_representatives_controller.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+module Veteran
+  module V0
+    class AccreditedRepresentativesController < ApplicationController
+      skip_before_action :authenticate
+      before_action :feature_enabled
+      before_action :verify_type
+      before_action :verify_sort
+      before_action :verify_long
+      before_action :verify_lat
+
+      DEFAULT_PAGE = 1
+      DEFAULT_PER_PAGE = 10
+      DEFAULT_SORT = 'distance_asc'
+
+      PERMITTED_TYPES = %w[attorney claim_agents organization].freeze
+
+      PERMITTED_ORGANIZATION_SORTS = %w[distance_asc distance_desc name_asc name_desc].freeze
+      PERMITTED_REPRESENTATIVE_SORTS = %w[distance_asc distance_desc first_name_asc first_name_desc last_name_asc
+                                          last_name_desc].freeze
+      def index
+        collection = Common::Collection.new(model_klass, data: accreditation_query)
+        resource = collection.paginate(**pagination_params)
+
+        render json: resource.data,
+               serializer: CollectionSerializer,
+               each_serializer: serializer_klass,
+               meta: resource.metadata
+      end
+
+      private
+
+      def organization_search?
+        @organization_search ||= search_params[:type] == 'organization'
+      end
+
+      def model_klass
+        @model_klass ||= if organization_search?
+                           'Veteran::Service::Organization'.constantize
+                         else
+                           'Veteran::Service::Representative'.constantize
+                         end
+      end
+
+      def serializer_klass
+        @serializer_klass ||= if organization_search?
+                                'Veteran::Accreditation::OrganizationSerializer'.constantize
+                              else
+                                'Veteran::Accreditation::RepresentativeSerializer'.constantize
+                              end
+      end
+
+      def base_query
+        @base_query ||= model_klass.find_within_max_distance(search_params[:long],
```

Comment:
Do we need to set this `@base_query` ivar? It looks like `#base_query` is only called once.